lib/purescript-cst/src/Language/PureScript/AST/Exported.hs

```diff
@@ -164,6 +167,9 @@ reorder refs =
     KindDeclaration _ _ n _ ->
       M.lookup (TyName n) refIndices <|> M.lookup (TyClassName (tyToClassName n)) refIndices
 
+    RoleDeclaration RoleDeclarationData{..} ->
+      M.lookup (TyName rdeclIdent) refIndices
+
     decl -> declName decl >>= flip M.lookup refIndices
```

Comment:
Similarly here. If `declName` included `RoleDeclaration`s, the above case isn't needed.